### ISSUE TITLE
Design Wizard component

### ISSUE THREAD

**Issue by joseegm** [SAP/fundamental#136]
### Description
The Wizard component let the user navigate a complicated flow, step by step. Permitting navigating back and forth between the steps.

### Requirements
1. The wizard component can be used page wide or inside a panel
  1.1 Page wide: the steps change the whole content of the page
  1.2 Inside a panel: the steps only change the content of the panel
    1.2.1 Can be displayed vertically as part of the page.
2. The wizard component should allow the user to navigate between the steps
  2.1 Should have "Next" and "Previous" buttons
  2.2 The steps are clickable, allowing the user to navigate directly to a specific step
3. The wizard component should display clearly the current step
  3.1 The steps can display a title
  3.2 The steps can display a order number
  3.3 The steps can display an error state
  3.4 The steps can be disabled or enabled
    3.4.1 The disabled steps should be visually represented and be different of the enabled steps

**Comment by joseegm** [comment]
@jeannewalters 
We can start looking at the requirements for this one

**Comment by jeannewalters** [comment]
I am going to go with a horizontal approach which is different than 1.2.1.

**Comment by joseegm** [comment]
1.2.1 Is there because we have a use case for one product:
https://app.zeplin.io/project/5a5e1cb96e836180d25dfebd/screen/5a8b1494e3f94d0d5d8445a1

I think is a very valid use case also for when they are many steps that doesn't fit horizontally.

**Comment by jeannewalters** [comment]
We would have to cover the many step scenario in the horizontal approach too.

**Comment by jeannewalters** [comment]
Here are the wires that were reviewed during the Experience call today.

Page Wizard (support responsive):
https://9p2ucw.axshare.com/#g=1&p=page_wizard_-_top
Page Wizard (support responsive): - 1st step:
https://9p2ucw.axshare.com/#g=1&p=page_wizard_-_first_step
Page Wizard - Last step (support responsive):
https://9p2ucw.axshare.com/#g=1&p=page_wizard_-_last_step
Modal Wizard (support responsive):
https://9p2ucw.axshare.com/#g=1&p=top_modal_wizard
Side Wizard (desktop ONLY):
https://9p2ucw.axshare.com/#g=1&p=desktop_page_wizard_-_side

**Comment by jeannewalters** [comment]
Here is the proposal:
**Top Page Wizard:**
Top wizard: https://k1xok8.axshare.com/#g=1&p=page_wizard_-_top
Error state: https://k1xok8.axshare.com/#g=1&p=page_wizard_-_error_state
First step: https://k1xok8.axshare.com/#g=1&p=page_wizard_-_first_step
Last step: https://k1xok8.axshare.com/#g=1&p=page_wizard_-_last_step

**Modal Wizard**
https://k1xok8.axshare.com/#g=1&p=top_modal_wizard

**Mobile Version** 
https://k1xok8.axshare.com/#g=1&p=mobile_wizard

**Desktop - Side Wizard**
https://k1xok8.axshare.com/#g=1&p=desktop_only_page_wizard_-_side

**Comment by LeoT7508** [comment]
@jeannewalters @xak @saadmhybris 

Wizard design sheet is here:

https://zpl.io/brrYG17

If there are any gaps I'll need the details so I can plug those in.

**Comment by jeannewalters** [comment]
@xak @saadmhybris - Zack, Saad suggested that we rename this component "Progress Indicator" which would be part of a wzard rather than calling it a wizard (which is really more like a starter page or floorplan. I would agree - especially since you added #429 yesterday which was already covered in the wires I attached to this ticket. I am going to make a note on #429 as well.  I've updated the ticket name for this one.

**Comment by jeannewalters** [comment]
@xak - Leo wants to run the use case of going back to previous steps past Remy for feedback and Remy is out until June 1. Should we move this to the next release?

**Comment by saadmhybris** [comment]
@xak the fiori wizard examples consists of the 'full' wizard with the progress bar, step heading, step content and next button. I agree on sticking with UI5 vocabulary and maybe we need to expand the scope of this component to include the full wizard elements/components?

**Comment by xak** [comment]
It should be noted that the wires attached above mix the component and the floorplan. For example, the vertical option has a description under the Step name which is not possible in a horizontal version. The latter shows the description in a panel which is where the floorplan context comes in. 

![screen shot 2018-05-30 at 11 28 00 am](https://user-images.githubusercontent.com/424119/40733944-8ef831b6-63fc-11e8-9671-e5667e559a84.png)

The progress nav itself must be very specific in what data is allowed. We can't easily create a single progress nav which can be horizontal or vertical and support all data options in either orientation. It should be clear which is best for what use case and we can also explain how the rest of page should be constructed.

## IMO
- The vertical placement is more appropriate for in-page content where a large number of items (all visible on the page) must be completed before saving the page. See #429. It's a handy reference that will stay fixed as the user scrolls through the content.
- The horizontal placement is better for a tab-like experience where one panel may influence the next panel. Steps that must be completed in sequence.

The UI5 implementation is interesting in that it becomes an in-page experience as you progress through the steps. You can scroll up or click on the steps to scroll back. In the branching example, an alert is shown if a change will affect downstream options.

**Comment by xak** [comment]
@jeannewalters I'm want to move this to 1.2 release. AFAIK no one is waiting for this component so it can take a lower priority. What do you think?

**Comment by jeannewalters** [comment]
@xak @saadmhybris The vertical approach also does not work in mobile and the horizontal approach does. The Rapid Retail team is implementing the wizard to set up their store. We are moving away from a wizard in the Data Visualization project. I believe that Fiore has both options and I've seen the vertical steps used as a wizard that moves the user between steps as well as more of a guide approach.  When I did the wires (before the Wizard was posted to Zeplin from Global Design), I expected it to be more of an all encompassing approach and not just a progress indicator. I agree with Saad that if this is all it is, we should call it Progress Indicator and create starter pages to support the suggested directions using multiple components for both a vertical and horizontal approach shown in the wires.
